### ISSUE TITLE
Build a combined Track and Submit flow on the FE, so we can use it to run our very first A/B test.

### ISSUE THREAD

**Issue by anmurali** [Expensify/App#47598]
**Problem**: The vast majority of signups are unengaged. Paid adoption from top or bottom up leads only happens when users are engaged with the product and derive value from it.

**Hypothesis**: The time it takes for a person to make a decision increases with the number and complexity of choices available. Presenting users with too many options can lead to decision fatigue, making the user experience overwhelming and less efficient.

**Test**: The Global Create menu is by far the most obvious and accessible option a new user explores to do something. Let’s make it a smaller, focused list that allows a user to pick one out of a very small set of choices. Instead of asking the user to decide between tracking and submitting an expense, let’s present a create expense that after the expense is manually entered or scanned allows the user to select what they want to do with that expense in the moment.

**Metric**: conversion in this test would be number of people that create an expense / number of people that see the create expense option Vs sum of the number of people that track or submit an expense / number of people that see track and submit as separate options in Global Create

**Notes**: We are already cleaning up the GC menu https://github.com/Expensify/App/issues/47274 so we focus the user on a smaller set of realistic actions. Once we do that, we should be ready to test an idea @shawnborton proposed, which is to use one single combined flow that allows the user to first create the expense and in the next step decide to either save it for now or submit to someone. 

Let's get started on building this and just put it on a beta for now and don't show it to anyone not on the beta. That way, when we finish cleaning up the GC and figure out how to set up the test, we can turn it on accordingly to the variant audience.
![image](https://github.com/user-attachments/assets/02f8b6b2-9fd4-4e24-b301-613362a56121)

- `Just track this for now` would basically mimic the track expense flow
- Choosing someone to submit it to would then follow the submit flow

**Comment by shawnborton** [comment]
Updated mock would look like this:
![CleanShot 2024-08-19 at 08 29 13@2x](https://github.com/user-attachments/assets/8c3bf24c-dd1c-401d-b777-d2f3fc2d8d1c)

cc @Expensify/design 

I think this would be a good place to start. Then I think separately we should really follow up and make some improvements to the tracked expense experience, to make it easier to send a tracked expense somewhere after you indicate that you want to save it.

**Comment by dubielzyk-expensify** [comment]
That looks good to me :)

**Comment by grgia** [comment]
I can help move this forward - will assign for now to track but need a bit more context

**Comment by grgia** [comment]
@anmurali @danielrvidal 
Curious how this would work. Is it % of all users? Or a subclass of users?

**Comment by anmurali** [comment]
@grgia how we actually deploy the test is still actively under discussion [here](https://expensify.slack.com/archives/C07HPDRELLD/p1724259376787309?thread_ts=1724257044.358639&cid=C07HPDRELLD). Cause we gotta adopt a framework that we can use repeatably over time. I think for now we could develop this variant, and put it behind a beta?

**Comment by grgia** [comment]
Yeah that works @anmurali. Do we have a doc or something this links to so I can get more context? 

Is the A test just the current FAB/Global Create menu?

**Comment by grgia** [comment]
I pushed a PR for the beta. Can I get confirmation that this is ready to be built so I can get one of our external agencies on it?

**Comment by anmurali** [comment]
I wrote up a problem/ hypothesis/ solution [here](https://expensify.slack.com/archives/C07HPDRELLD/p1724326134699559)

**Comment by anmurali** [comment]
We are ready to build [this](https://github.com/Expensify/App/issues/47598#issuecomment-2296460075) and put on a beta while we figure out how we will setup testing itself.

**Comment by grgia** [comment]
Reached out to callstack and @fabioh8010 has offered to take this on - See here https://expensify.slack.com/archives/C03UK30EA1Z/p1724426198287819

**Comment by grgia** [comment]
@shawnborton could you update the main description with the final mock?

**Comment by grgia** [comment]
@fabioh8010 this is the beta we will build under https://github.com/Expensify/App/pull/47921

**Comment by fabioh8010** [comment]
Hi, I'm Fábio - expert agency contributor - and I would like to work on this issue.

**Comment by fabioh8010** [comment]
@grgia @anmurali Do you mind sharing the content of these threads ([here](https://expensify.slack.com/archives/C07HPDRELLD/p1724259376787309?thread_ts=1724257044.358639&cid=C07HPDRELLD), [here](https://expensify.slack.com/archives/C07HPDRELLD/p1724326134699559))? I don't have access to them. Thanks!

**Comment by anmurali** [comment]
https://www.figma.com/design/ReBRDcqMXgtvPgWBBCmBxs/Wave-9%3A-Track-Expense?node-id=512-11640&t=0d1PZheynnl6iKNr-1

I updated the issue description with what's in the threads. No need to read them! :)

**Comment by danielrvidal** [comment]
Rather than saying `Just track this for now` I think we could/should use `Personal tracking` or `Track for myself`.  My reasoning is 1) it's shorter and 2) I think it's more clear because just track this for now could still be submitted to someone else. 

@anmurali @shawnborton @dubielzyk-expensify @jamesdeanexpensify not sure if you agree or like what it currently is more. 

![image](https://github.com/user-attachments/assets/4da253ed-d209-4634-b68a-db2322d00378)

**Comment by shawnborton** [comment]
Personally I think we need some kind of verbiage that makes it clear that you are not trying to submit it, so saying "Just track it for now" reads as though you are acknowledging that you are on a submit screen, but you don't want to submit it. Actually my original suggestion was "Don't send this, just track for now" to be extra clear here but I think someone suggested we shorten that. 

I don't feel too strongly though, happy to go with whatever you all think is best.

**Comment by jamesdeanexpensify** [comment]
'Just track, don't send (for now)' we could even remove the '(for now)' but I like keeping the affirmative in front.

**Comment by jamesdeanexpensify** [comment]
I would also be fine leaving 'Just track this for now' though.

**Comment by danielrvidal** [comment]
Sounds like people like the affirmative so far so happy to stick with what we have right now. It just caught my eye.

**Comment by shawnborton** [comment]
Nice, I still think I like adding the `don't` in there somewhere to be extra clear, so I think something like `Just track it, don't send` would be my vote.

**Comment by jamesdeanexpensify** [comment]
I see the [Figma](https://www.figma.com/design/ReBRDcqMXgtvPgWBBCmBxs/Wave-9%3A-Track-Expense?node-id=512-11660&t=e3DIQrVFFBWjqUi8-0) file and I asked for edit access. We could look at a few options to see what looks nice?

**Comment by jamesdeanexpensify** [comment]
![image](https://github.com/user-attachments/assets/28ca1cac-9151-4566-b6a4-a98fcb591dcc)

**Comment by dubielzyk-expensify** [comment]
Gave you acces @jamesdeanexpensify 👍 

I actually really like "Track for myself" cause I feel like this whole Track vs Submit thing is such "Expensify language". I especially like that it's highlighting "me" or "personal".

**Comment by shawnborton** [comment]
True. I'm mostly just thinking that we want something that is clearly in opposition to the title of the page, which is "Send to someone"

**Comment by dubielzyk-expensify** [comment]
Agree with that Mr Borton

**Comment by quinthar** [comment]
Can we make the title "Submit expense" so we follow the overall language of the product, and not introduce some new concept of "sending an expense to someone"?

**Comment by shawnborton** [comment]
Sure, that would look like this:
![CleanShot 2024-08-27 at 08 19 44@2x](https://github.com/user-attachments/assets/e1cd1af4-9e8a-4cd9-8347-fc962554f3f3)

**Comment by grgia** [comment]
@shawnborton @jamesdeanexpensify 
Could you ping me when we've confirmed the final mock up so I can keep this moving? Thank you!

**Comment by jamesdeanexpensify** [comment]
I would be fine with [this](https://github.com/Expensify/App/issues/47598#issuecomment-2312411510)!

**Comment by shawnborton** [comment]
Yup, same!

**Comment by fabioh8010** [comment]
Update: Working on the proposal / action plan to refactor and develop the combined flow. I plan to post it here tomorrow.

**Comment by fabioh8010** [comment]
Update: Validating / polishing the final proposal after encountering some problems.

**Comment by fabioh8010** [comment]
@grgia 

# Proposal

**1)** Change [FloatingActionButtonAndPopover](https://github.com/Expensify/App/blob/main/src/pages/home/sidebar/SidebarScreen/FloatingActionButtonAndPopover.tsx) to pass different data to `PopoverMenu`'s `menuItems` prop if the beta is enabled. If the beta is enabled we should render the `Create expense` option instead of `Track expense` and `Submit expense`.

**2)** Create a new `create` IOU Type in [CONST.ts](https://github.com/Expensify/App/blob/main/src/CONST.ts). This new type will represent the new global `Create expense` flow.

**3)** Modify [IOUUtils.isValidMoneyRequestType()](https://github.com/Expensify/App/blob/main/src/libs/IOUUtils.ts) to account for the new type (maybe we can also check if user has beta here), so it will allow [AccessOrNotFoundWrapper](https://github.com/Expensify/App/blob/main/src/pages/workspace/AccessOrNotFoundWrapper.tsx) to let the user access the new flow when accessing the [IOURequestStartPage](https://github.com/Expensify/App/blob/main/src/pages/iou/request/IOURequestStartPage.tsx) page. Additionaly, a new header title will be added for `IOURequestStartPage`.

**4)** For each tab (Manual `IOURequestStepAmount`, Scan `IOURequestStepScan`, Distance `IOURequestStepDistance`), modify the code to navigate the user to [IOURequestStepParticipants](https://github.com/Expensify/App/blob/main/src/pages/iou/request/step/IOURequestStepParticipants.tsx) page if the IOU Type is `create`.

**5)** In `IOURequestStepParticipants` page, modify [MoneyRequestParticipantsSelector](https://github.com/Expensify/App/blob/main/src/pages/iou/request/MoneyRequestParticipantsSelector.tsx) component to add the `Just track it` button (using `MenuItem`) to its `SelectorList`'s `headerContent` prop if the IOU Type is `create`. 

This `MenuItem`, when clicked, will call the new `onTrackExpensePress` prop that will be used in `IOURequestStepParticipants` to navigate the user to [IOURequestStepConfirmation](https://github.com/Expensify/App/blob/main/src/pages/iou/request/step/IOURequestStepConfirmation.tsx) page passing the `track` IOU Type, **thus allowing the user to finish the flow as they was tracking an expense**.

In case the user select a participant, they will be navigated to `IOURequestStepConfirmation` passing the `submit` IOU Type instead, **so the user can finish the flow as they was submitting an expense**.

---

With this proposal I believe we can achieve this goal with minimal changes in the code while having support for both `Create expense` and  `Track expense` / `Submit expense` options.

There are other minor changes that will be required but the main ones are these. I believe all this can be done in one PR because there isn't too much work / refactoring to do as I noted in my investigations and tests.

---

@grgia I have one question though: When we click the `Track expense` button from Global Create, we are firstly presented with a Track onboarding modal that includes a video.

**1)** Should we show them when the user navigates to the confirmation page (after clicking the `Just track it` button)?

**2)** If yes, maybe we would need to have a new video for this modal as the current one won't make sense?

**Comment by shawnborton** [comment]
@fabioh8010 I think for this test, it would make sense to just remove the video from the combined flow entirely? Curious what @anmurali thinks too.

**Comment by fabioh8010** [comment]
Yeah I don't think we need the video modal there, it looks redundant to me.

**Comment by anmurali** [comment]
Yes, let's remove that for sure.

**Comment by fabioh8010** [comment]
@grgia bump 😄

**Comment by grgia** [comment]
@fabioh8010 Proposal looks good, thank you! Would you update it to specify that we will remove the video from that flow?

**Comment by grgia** [comment]
**Progress Update**

- I chatted with @fabioh8010, and implementation will start tomorrow. 
- I created an issue [here](https://github.com/Expensify/App/issues/48787) for the feature implementation

We should have the feature ready to go soon for when we decide how we want to A/B test

cc @anmurali

**Comment by fabioh8010** [comment]
> @fabioh8010 Proposal looks good, thank you! Would you update it to specify that we will remove the video from that flow?

Updated!

**Comment by anmurali** [comment]
@fabioh8010 - responding to your question [here](https://github.com/Expensify/App/pull/49007#issuecomment-2368747675)

<img width="546" alt="image" src="https://github.com/user-attachments/assets/f5bb5e7b-2689-4a2d-b980-0c4d0b40c26c">

<img width="574" alt="image" src="https://github.com/user-attachments/assets/75b6c3d6-75d2-4250-8f84-d3b782cded82">


<img width="531" alt="image" src="https://github.com/user-attachments/assets/b68f04ba-fd47-4faa-826a-2fe1fbba405c">

**Comment by dubielzyk-expensify** [comment]
Sorry if I'm coming in late here, but just curious what are we actually A/B testing here? Is it changing the wording from `Submit expense` to `Create expense`?

**Comment by anmurali** [comment]
Submit/ Track as separate options vs. Create. And where there is only one or the other of the former - that vs. Create.

**Comment by grgia** [comment]
@anmurali we've merged the feature. Once it's deployed and passes QA I can merge the code releasing it to half of users

**Comment by anmurali** [comment]
@grgia - We also need to tweak the onboarding tasks for the test audience.

Track an expense task

> **Title** Track an expense
> **Description** Track an expense in any currency, whether you have a receipt or not.
> 
> Here’s how to track an expense:
> 
> 1. Click the green + button.
> 2. Choose Create expense.
> 3. Enter an amount or scan a receipt.
> 4. Click "Just track it (don't submit it)".
> 5. Click Track.
> 
> And you’re done! Yep, it’s that easy.

Submit an expense task

> **Title**: Submit an expense
> **Description**: Submit an expense by entering an amount or scanning a receipt.
> 
> Here’s how to submit an expense:
> 
> 1. Click the green + button.
> 2. Choose Create expense.
> 3. Enter an amount or scan a receipt.
> 4. Add your reimburser to the request.
> 
> Then, send your request and wait for that sweet “Cha-ching!” when it’s complete.

cc @jamesdeanexpensify for a review

**Comment by jamesdeanexpensify** [comment]
Looks good to me!

**Comment by trjExpensify** [comment]
I agree, that looks weird.

**Comment by dannymcclain** [comment]
The only reason I could see keeping it is if we think people might search for themselves... But that does not seem like a compelling reason, so I'm down to ditch it once someone starts searching.

**Comment by shawnborton** [comment]
Sweet, I say we ditch it then. @grgia thoughts?

**Comment by dubielzyk-expensify** [comment]
Agree. I guess if we wanted to be clever we could show "Just track it (don't submit it)" if the search query matches the name of the user? On that note, sorry if this has been raised, what happens if I search for my own name here? Cause sending it to "Jon (you)" and clicking "Just track it" is the same action, right?

**Comment by shawnborton** [comment]
> Cause sending it to "Jon (you)" and clicking "Just track it" is the same action, right?

Yup, that's right. Though thinking back on this, maybe we want to pause any changes here and see how @trjExpensify 's proposal [here](https://expensify.slack.com/archives/C07HPDRELLD/p1731690049777719?thread_ts=1731604883.924969&cid=C07HPDRELLD) goes?

**Comment by danielrvidal** [comment]
Yup, I was going to say I like that update but I think we should pause as @trjExpensify is going to be proposing that update.

**Comment by grgia** [comment]
Yeah, I think this last discussion landed in a good place. I agree that it's confusing to have both
